### DISCUSSION TITLE
Read_write test error

### DISCUSSION THREAD

**Discussion by SeahorseRTHK** [Bodmer/TFT_eSPI#1876] [Compatible displays and setup files]
Hello, first post ever! I did the read write test from the examples (TFT_ReadWrite_Test) (didn't edit anything) and all the results came out as an error, read = 2104. Any idea how to fix that? I'm not sure but I think my User_Setup.h settings are off. I'm using wt-86-32-3zw1 by Wireless-Tag, Arduino IDE, FTDI.
These settings are in my User_Setup.h file:
```
#define ILI9488_DRIVER 
#define TFT_WIDTH 320
#define TFT_HEIGHT 320
#define TFT_BCKL 0 
#define TFT_CS 3   
#define TFT_DC 38 
#define TFT_RST 37
#define TFT_WR 39 
#define TFT_RD 2
#define TFT_D0 36
#define TFT_D1 35
#define TFT_D2 34
#define TFT_D3 33
#define TFT_D4 21
#define TFT_D5 18
#define TFT_D6 17
#define TFT_D7 16
#define ESP32_PARALLEL
#define TOUCH_SDA 15
#define TOUCH_SCL 14
#define TOUCH_IRQ 13
#define TOUCH_RST 12
#define TOUCH_DRIVER 0x1680
#define TFT_ROTATION 6
#define LGFX_USE_V1 1
#define SUPPORT_TRANSACTIONS
#define SPI_TOUCH_FREQUENCY 2500000
```
I think I don't know how to properly set it up, I got these settings from [https://github.com/HASwitchPlate/openHASP/blob/master/user_setups/esp32s2/wt-86-32-3zw1.ini](url)
This is the serial output:
![Capture](https://user-images.githubusercontent.com/94155812/173481654-70100a51-47af-4a37-ad81-6a77647b1c89.JPG)

**Comment by Bodmer** [comment]
Here is an example setup:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup14_ILI9341_Parallel.h

Note the comments about the pin number ranges that are supported.

Some of the pins are input only! See this reference guide:
https://randomnerdtutorials.com/esp32-pinout-reference-gpios/
